ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/TransactionReceipt.java

```diff
@@ -191,23 +202,27 @@ public static TransactionReceipt readFrom(
       final LogsBloomFilter bloomFilter = LogsBloomFilter.readFrom(input);
       final List<Log> logs = input.readList(Log::readFrom);
       final Optional<Bytes> revertReason;
+      long gasRemaining = -1L;
       if (input.isEndOfCurrentList()) {
         revertReason = Optional.empty();
       } else {
-        if (!revertReasonAllowed) {
+        if (!metadataAllowed) {
           throw new RLPException("Unexpected value at end of TransactionReceipt");
         }
         revertReason = Optional.of(input.readBytes());
+        gasRemaining = input.readLong();
```

Comment:
What about when we do a migration?  What if we have an old DB with revert reason but not miner data and start extracting it?  gasRemaining may not always be there.

I would wrap everything read after the `metadataAllowed` check in `if (!input.isEndOfCurrentList()) { ... }` individually as a safety measure.  So two times in this case.

Reply:
Will wrap the `gasRemaining` but the block starts with the check so the `revertReason` read is safe.